packages/boot-contract-derive/src/lib.rs

```diff
@@ -52,12 +59,149 @@ pub fn contract(attrs: TokenStream, input: TokenStream) -> TokenStream {
             type MigrateMsg = #migrate;
         }
     );
     struct_def.into()
 }
+
+#[proc_macro_attribute]
+pub fn boot_contract(_attrs: TokenStream, mut input: TokenStream) -> TokenStream {
+    let cloned = input.clone();
+    let mut item = parse_macro_input!(cloned as syn::Item);
+
+    let Item::Fn(boot_func) = &mut item else {
+        panic!("Only works on functions");
+    };
+
+    // Now we get the fourth function argument that should be the instantiate message
+    let signature = &mut boot_func.sig;
+    let func_ident = signature.ident.clone();
+    let func_type = get_func_type(signature);
+
+    let message_idx = match func_ident.to_string().as_ref() {
+        "instantiate" | "execute" => 3,
+        "query" | "migrate" => 2,
+        _ => panic!("Function name not supported for the macro"),
+    };
+
+    let message = match signature.inputs[message_idx].clone() {
+        FnArg::Typed(syn::PatType { ty, .. }) => *ty,
+        _ => panic!("Only typed arguments"),
+    };
+
+    let wasm_name = get_wasm_name();
+    let name = get_crate_to_struct();
+
+    let struct_def = quote!(
+            #[derive(
+                ::std::clone::Clone,
+            )]
+            pub struct #name<Chain: ::boot_core::CwEnv>(::boot_core::Contract<Chain>);
+
+            impl<Chain: ::boot_core::CwEnv> ::boot_core::ContractInstance<Chain> for #name<Chain> {
+                fn as_instance(&self) -> &::boot_core::Contract<Chain> {
+            &self.0
+        }
+            fn as_instance_mut(&mut self) -> &mut ::boot_core::Contract<Chain> {
+                &mut self.0
+            }
+        }
+
+        fn find_workspace_dir() -> ::std::path::PathBuf{
+            let crate_path = env!("CARGO_MANIFEST_DIR");
+            let mut current_dir = ::std::path::PathBuf::from(crate_path);
+            match find_workspace_dir_worker(&mut current_dir) {
+                Some(path) => path,
+                None => current_dir,
+            }
+        }
+
+        fn find_workspace_dir_worker(dir: &mut::std::path::PathBuf) -> Option<::std::path::PathBuf> {
+            loop {
+                // First we pop the dir
+                if !dir.pop() {
+                    return None;
+                }
+                let cargo_toml = dir.join("Cargo.toml");
+                if ::std::fs::metadata(&cargo_toml).is_ok() {
+                    return Some(dir.clone());
+                }
+            }
+        }
+
+        // We add the contract creation script
+        impl<Chain: ::boot_core::CwEnv> #name<Chain> {
+            pub fn new(contract_id: &str, chain: Chain) -> Self {
+
+                // We get the workspace dir
+                let workspace_dir = find_workspace_dir();
+
+                // We build the artififacts from the artifacts folder (by default) of the package
+                let file_path = &format!("{}/artifacts/{}.wasm", workspace_dir.display().to_string(), #wasm_name);
```

Comment:
when compiling with the arm optimizer the artifacts get a post-fix of `-aarch64.wasm`. Since a lot of devs use apple macbooks we should try to account for it.

Reply:
Have a PR open for something similar: https://github.com/Abstract-OS/BOOT/pull/78